### PR TITLE
2743 discard pushed back application

### PR DESCRIPTION
**Pull request by serge-gaia**
* Applicants should be able to discard an application that has been pushed back to them by an officer.  
* The discarded application has a new status (customer and processing) 'discarded'.  
* Discarded app should not appear in the customer dashboard.
See spec:
[#2743](https://kanboard.dpaw.wa.gov.au/?controller=TaskViewController&action=show&task_id=2743&project_id=24)

### CODE REVIEW

File: wildlifelicensing/apps/applications/tests/test_entry.py [hunk 1]
```diff
@@ -428,3 +429,109 @@ def test_user_not_logged_is_redirected_to_login(self):
...
+        application = helpers.create_and_lodge_application(self.applicant)
+        self.assertNotIn(application.processing_status, Application.CUSTOMER_DISCARDABLE_STATE)
+        # officer request amendment
+        url = reverse('wl_applications:amendment_request')
+        self.client.login(self.officer.email)
+        resp = self.client.post(url, data={
+            'application': application.pk,
+            'officer': self.officer.pk,
+            'reason': 'missing_information',
+            'text': 'You cannot kill wombats!'
```
Comment (tony-gaia) [inline wildlifelicensing/apps/applications/tests/test_entry.py]: wtf?

File: wildlifelicensing/apps/applications/tests/test_entry.py [hunk 2]
```diff
@@ -428,3 +429,109 @@ def test_user_not_logged_is_redirected_to_login(self):
...
+        self.assertTrue('action_url' in resp.context)
+        self.assertEquals(resp.context['action_url'], url)
+        application.refresh_from_db()
+        # status should be unchanged
+        self.assertNotEqual(application.processing_status, 'discarded')
+        self.assertNotEqual(application.customer_status, 'discarded')
+        self.assertEqual(application.processing_status, previous_processing_status)
+        self.assertEqual(application.customer_status, previous_customer_status)
+
+        # post! Like clicking on confirm
```
Comment (tony-gaia) [inline wildlifelicensing/apps/applications/tests/test_entry.py]: Yes, we know what post is.

Comment (leftclickben) [reply wildlifelicensing/apps/applications/tests/test_entry.py]: lol :+1:

File: wildlifelicensing/apps/applications/views/entry.py [hunk 3]
```diff
@@ -312,7 +359,7 @@ def form_valid(self, form):
         application.applicant.save()
 
         # update any other applications for this user that are awaiting ID upload
-#       for application in Application.objects.filter(applicant_profile__user=applicant):
+        #       for application in Application.objects.filter(applicant_profile__user=applicant):
```
Comment (tony-gaia) [inline wildlifelicensing/apps/applications/views/entry.py]: remove commented code

File: wildlifelicensing/apps/applications/views/entry.py [hunk 4]
```diff
@@ -515,13 +562,13 @@ def get_context_data(self, **kwargs):
             return redirect('wl_applications:new_application')
 
         kwargs['is_payment_required'] = not is_licence_free(generate_product_title(application)) and \
-            not application.invoice_reference and is_customer(self.request.user)
+                                        not application.invoice_reference and is_customer(self.request.user)
```
Comment (tony-gaia) [inline wildlifelicensing/apps/applications/views/entry.py]: You only do this sort of indenting when the code is within parentheses of some sort. Same for below on line 641

File: wildlifelicensing/apps/applications/tests/test_entry.py (hunk 2 above)
Comment (tony-gaia) [reply wildlifelicensing/apps/applications/tests/test_entry.py]: Erm, sir, I think you're in the wrong house.

Comment (serge-gaia) [reply wildlifelicensing/apps/applications/tests/test_entry.py]: What are you doing here Ben? :-)

Comment (leftclickben) [reply wildlifelicensing/apps/applications/tests/test_entry.py]: eye_of_sauron

Comment (leftclickben) [reply wildlifelicensing/apps/applications/tests/test_entry.py]: I am subscribed to notifications from the gaiaresources organisation :-)

File: wildlifelicensing/apps/applications/views/entry.py (hunk 4 above)
Comment (serge-gaia) [reply wildlifelicensing/apps/applications/views/entry.py]: Jetbrains auto-format is not perfect

File: wildlifelicensing/apps/applications/views/entry.py (hunk 3 above)
Comment (serge-gaia) [reply wildlifelicensing/apps/applications/views/entry.py]: The line was already there. It's just my auto-format that changed it.
Will remove.

**Pull request merged**